### PR TITLE
Release Pipeline

### PR DESCRIPTION
**Pull request by jpoon**
On a successful build of the `release` branch, Travis will:

1) Create a GitHub release uploading the packaged vsix
2) Publish the new version to the marketplace

Moving forward, for pushing a release, it will be:

1) merge `master` to `release` branch
2) `git tag`/`gulp release` and `git push origin --tags`

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,37 +1,61 @@
...
+    - xclip
+
+language: node_js
+
+node_js:
+- 6.3.1
+
+env:
+  global:
+    secure: F3+6ObO8G2XDTXXSWizxdV4YzjWWwFKfISwYnKTwnAy4LXLPzcHnrh31rFmKeXSddZmWiVMhy++a4Y9IOTi0DrSnUYaaIP35zMIvMYIqdSa6v3b93Z0fwZmniXe0e4EBIhcdnMOz1fdxGhd0oWP/YyywOm32TjymhqbApT9AzqUodpGqRmA30B1tOFr09K+A2EiIr+4QGsB/a3VUv24zWWUHtkXaJ6tQ7rqt6944tGK4RtjAIzXLoePjJ+kmTqY6fMk/sPmYxHXnzjaEsIFBd5dYDCFtjR0oqjdq2OXc+bwIjl4t3Wh4RwE8bUEcUSnkPRqlJH5Hni5vfPRirb4qH4uoFmLNa8Pg2dPI44cYBvWVGSJ5GqXD5jV64waR3CM3VC2Ll76KWdlqdUo0cNIV75ul+UdieuNghTnw3OTYIWyhvrLqfnabOG81z09brNapZuCPj3rZ+sjTWf7beRoH877UGH0C6jimxNk0syjGL0+pXH0UR0dI+7RWaNyPAMfvjz3m9UpDw3NtdDGNMT1D3VgZz+INwU6XMUSpg36S1BzMTfi6NJm+srA5BDHyC/MuYjO+l/1gaj0+dvkpMyyj9btotYFmfemEFlm2AHGMkITaucUrmCQKIM9ZL7l/fZfvPkUGbaaLH7EONL7KLLaj6pk2z7na91/3I1pWg4yJDPE=
```
Comment (rebornix) [inline .travis.yml]: Not quite sure but is it possible to put the key in Travis env instead of here?

Comment (jpoon) [inline .travis.yml]: I think they would have the same effect.

I would prefer putting it in this manner as then our configs are in one central location and not scattered about.

**Comment by jpoon** [comment]
@johnfn I simplified the release workflow removing the need to manage a `release` branch. Added instructions to the contributing.md.

In order to push a new release, all that is needed is to create a git tag. there's a helper function provided in the gulpfile for doing so: `gulp patch|minor|major` which will rev the package.json version as well as apply a git tag.

Travis will then:
1) build
2) test
3) publish the new version to the marketplace
4) create a GitHub release uploading the generated vsix

**Comment by johnfn** [comment]
LGTM. Thanks for maintaining our infra, @jpoon! It's a big help. :)

**Pull request merged**